src/include/misc.i

```diff
@@ -223,3 +223,24 @@ __wt_txn_context_check(WT_SESSION_IMPL *session, bool requires_txn)
 		    session->name);
 	return (0);
 }
+
+/*
+ * __wt_state_yield_sleep --
+ *	Sleep while waiting, after a thousand yields.
+ */
+static inline void
+__wt_state_yield_sleep(uint64_t *yield_count, uint64_t *sleep_count)
+{
+	/*
+	 * We yield before retrying, and if we've yielded enough times, start
+	 * sleeping so we don't burn CPU to no purpose.
+	 */
+	if ((*yield_count) < WT_THOUSAND) {
+		(*yield_count)++;
+		__wt_yield();
+		return;
+	}
+
+	(*sleep_count) = WT_MIN((*sleep_count) + 100, WT_THOUSAND);
```

Comment:
I know this code has just been moved here, but what's the point of this line?  We're going to count up to 1000 yields, then after that we'll keep setting `sleep_count` to `min(1100, 1000)`, aka 1000.

Was this code originally intended to do something smarter?

Reply:
We sleep for 100, 200, 300 ... 900 usecs, and from then on, 1000 usecs?

The sleep counter value is also fed back into the statistics counter of how long we slept, but I don't see any functionality there.

Reply:
Sorry, my mistake -- I conflated `sleep_count` and `yield_count`.